src/events/event.rs

```diff
@@ -9,12 +9,16 @@ use crate::{
     events::warehouse_events::{Empty, WarehouseEvent},
 };
 use candid::Principal;
+use hyper::client::HttpConnector;
 use ic_agent::Agent;
 use log::{error, info};
 use reqwest::Client;
 use serde_json::Value;
+use tokio::sync::OnceCell;
 use std::time::Duration;
-use yup_oauth2::ServiceAccountAuthenticator;
+use yup_oauth2::{authenticator::Authenticator, hyper_rustls::HttpsConnector, ServiceAccountAuthenticator};
+
+static AUTH: OnceCell<Authenticator<HttpsConnector<HttpConnector>>> = OnceCell::const_new();
```

Comment:
Can we use AppState instead since WarehousEventsService already has shared_state. init_auth can become part of the same. Check app_state.rs

Reply:
Reason is to maintain consistency of doing things

Reply:
just fixed this, code required a LOT of moving around. please verify the current logic once, since you wrote it initially and know how everything ties together @komal-rs #44